# Retail Sales Analysis

## Data Understanding

### Business Objective

The purpose of this project is to analyze retail sales data to identify sales patterns, customer behavior, regional performance, and profitability trends. The insights obtained will support business decision-making and serve as the foundation for an interactive Power BI dashboard.

In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Display all columns
pd.set_option("display.max_columns", None)

# Display numbers with two decimal places
pd.options.display.float_format = "{:.2f}".format

In [9]:
# Load the dataset
df = pd.read_csv("../data/raw/Superstore.csv")

# Display the first five rows
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2017-152156,2017-11-08,2017-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.00,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.96,2,0.00,41.91
1,2,CA-2017-152156,2017-11-08,2017-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420.00,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.94,3,0.00,219.58
2,3,CA-2017-138688,2017-06-12,2017-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036.00,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.62,2,0.00,6.87
3,4,US-2016-108966,2016-10-11,2016-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.00,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.58,5,0.45,-383.03
4,5,US-2016-108966,2016-10-11,2016-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311.00,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.37,2,0.20,2.52


In [10]:
df.shape

(9994, 21)

In [11]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   str    
 2   Order Date     9994 non-null   str    
 3   Ship Date      9994 non-null   str    
 4   Ship Mode      9994 non-null   str    
 5   Customer ID    9994 non-null   str    
 6   Customer Name  9994 non-null   str    
 7   Segment        9994 non-null   str    
 8   Country        9994 non-null   str    
 9   City           9994 non-null   str    
 10  State          9994 non-null   str    
 11  Postal Code    9983 non-null   float64
 12  Region         9994 non-null   str    
 13  Product ID     9994 non-null   str    
 14  Category       9994 non-null   str    
 15  Sub-Category   9994 non-null   str    
 16  Product Name   9994 non-null   str    
 17  Sales          9994 non-null   float64
 18  Quantity       9994

In [12]:
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.00,9983.00,9994.00,9994.00,9994.00,9994.00
mean,4997.50,55245.23,229.86,3.79,0.16,28.66
std,2885.16,32038.72,623.25,2.23,0.21,234.26
min,1.00,1040.00,0.44,1.00,0.00,-6599.98
25%,2499.25,23223.00,17.28,2.00,0.00,1.73
50%,4997.50,57103.00,54.49,3.00,0.20,8.67
75%,7495.75,90008.00,209.94,5.00,0.20,29.36
max,9994.00,99301.00,22638.48,14.00,0.80,8399.98


In [13]:
df.describe(include="object")

C:\Users\emili\AppData\Local\Temp\ipykernel_10636\702825166.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include="object")


,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Region,Product ID,Category,Sub-Category,Product Name
count,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994,9994
unique,5009,1236,1334,4,793,793,3,1,531,49,4,1862,3,17,1850
top,CA-2018-100111,2017-09-05,2016-12-16,Standard Class,WB-21850,William Brown,Consumer,United States,New York City,California,West,OFF-PA-10001970,Office Supplies,Binders,Staple envelope
freq,14,38,35,5968,37,37,5191,9994,915,2001,3203,19,6026,1523,48


## Initial Findings

The dataset contains **9,994 records** and **21 variables**, providing sufficient information for a comprehensive retail sales analysis.

The initial inspection reveals that most columns are complete, although the **Postal Code** column contains **11 missing values**, which will be addressed during the data cleaning phase.

Additionally, the **Order Date** and **Ship Date** columns are currently stored as text (`str`) and will need to be converted to datetime format to enable time-based analysis.

The dataset includes numerical variables such as **Sales**, **Profit**, **Quantity**, and **Discount**, which will be used to calculate business KPIs and evaluate sales performance.

Categorical variables provide detailed information about customers, products, regions, and shipping methods, enabling segmentation and profitability analysis.

## Data Quality Assessment

Before performing any analysis, it is essential to evaluate the quality of the dataset by identifying missing values, duplicate records, and potential inconsistencies.

In [14]:
# Check missing values
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country           0
City              0
State             0
Postal Code      11
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
Quantity          0
Discount          0
Profit            0
dtype: int64

In [15]:
# Percentage of missing values
missing_percentage = (df.isnull().sum() / len(df)) * 100

missing_percentage

Row ID          0.00
Order ID        0.00
Order Date      0.00
Ship Date       0.00
Ship Mode       0.00
Customer ID     0.00
Customer Name   0.00
Segment         0.00
Country         0.00
City            0.00
State           0.00
Postal Code     0.11
Region          0.00
Product ID      0.00
Category        0.00
Sub-Category    0.00
Product Name    0.00
Sales           0.00
Quantity        0.00
Discount        0.00
Profit          0.00
dtype: float64

In [16]:
df.duplicated().sum()

np.int64(0)

In [17]:
df.nunique()

Row ID           9994
Order ID         5009
Order Date       1236
Ship Date        1334
Ship Mode           4
Customer ID       793
Customer Name     793
Segment             3
Country             1
City              531
State              49
Postal Code       630
Region              4
Product ID       1862
Category            3
Sub-Category       17
Product Name     1850
Sales            5825
Quantity           14
Discount           12
Profit           7287
dtype: int64

## Data Quality Summary

The data quality assessment indicates that the dataset is generally clean and suitable for analysis.

Only the **Postal Code** column contains missing values, representing approximately **0.11%** of the dataset. Since this percentage is very small, these records are not expected to significantly affect the analysis.

No duplicate records were identified, indicating that each observation represents a unique transaction.

Additionally, the dataset contains a diverse set of categorical variables, including **793 unique customers**, **531 cities**, **49 states**, **1,850 products**, and **17 product sub-categories**, providing a rich foundation for segmentation and business analysis.

In [18]:
# Display data types
df.dtypes

Row ID             int64
Order ID             str
Order Date           str
Ship Date            str
Ship Mode            str
Customer ID          str
Customer Name        str
Segment              str
Country              str
City                 str
State                str
Postal Code      float64
Region               str
Product ID           str
Category             str
Sub-Category         str
Product Name         str
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object

## Data Types Assessment

The dataset contains appropriate numerical and categorical data types for most variables.

However, the **Order Date** and **Ship Date** columns are currently stored as text (`object`) instead of datetime format. These columns will be converted during the data cleaning phase to enable time-series analysis and date-based calculations.